src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -1353,26 +1401,27 @@ void AggregateEntityResourceVaultInternal(long entityId, long resourceEntityId,
                         GlobalEntityId = e.ReferencedVault.DatabaseGlobalAncestorId,
                         ResourceEntityId = e.ReferencedResource.DatabaseId,
                         VaultEntityId = e.ReferencedVault.DatabaseId,
-                        IsRoyaltyVault = e.ReferencedVault.GetDatabaseEntity<VaultEntity>().IsRoyaltyVault,
+                        IsRoyaltyVault = e.ReferencedVault.GetDatabaseEntity<InternalFungibleVaultEntity>().IsRoyaltyVault,
                         Balance = e.Balance,
                     };
                 })
                 .ToList();
 
             var entityNonFungibleVaultHistoryToAdd = nonFungibleVaultChanges
+                .GroupBy(x => new { x.StateVersion, x.ReferencedVault, x.ReferencedResource })
```

Comment:
I'm just thinking if we shouldn't use this "previous + mostRecentXxx" pattern. Are we sure we can operate only on data seen in TX stream? i.e. don't we need previous record form database?